lib/libzfs/Makefile.am

```diff
@@ -61,7 +61,7 @@ libzfs_la_LIBADD = \
 	$(top_builddir)/lib/libzfs_core/libzfs_core.la \
 	$(top_builddir)/lib/libzutil/libzutil.la
 
-libzfs_la_LIBADD += -lm $(LIBSSL)
+libzfs_la_LIBADD += -lm ${BSDLIB} $(LIBSSL)
```

Comment:
Something is breaking userland builds when "libbsd-dev" is installed (output from my Debian8 builder):

```
make[3]: Entering directory '/usr/src/zfs/lib/libicp'
  CC       spi/kcf_spi.lo
In file included from ../../include/sys/zfs_context.h:92:0,
                 from ../../module/icp/spi/kcf_spi.c:33:
../../lib/libspl/include/umem.h: In function 'umem_cache_create':
../../lib/libspl/include/umem.h:150:3: error: implicit declaration of function 'strlcpy' [-Werror=implicit-function-declaration]
   strlcpy(cp->cache_name, name, UMEM_CACHE_NAMELEN);
   ^
cc1: all warnings being treated as errors
Makefile:1004: recipe for target 'spi/kcf_spi.lo' failed
make[3]: *** [spi/kcf_spi.lo] Error 1
make[3]: Leaving directory '/usr/src/zfs/lib/libicp'
Makefile:608: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/usr/src/zfs/lib'
Makefile:793: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/usr/src/zfs'
Makefile:661: recipe for target 'all' failed
make: *** [all] Error 2
```

Reply:
My guess is that this is because `libbsd.so` provides a lot of the same functionality as `libspl.so` and when both are included we run in conflicts.  Assuming `libbsd.so` is provided by all the major distributions, and provides what we need, it would be interesting to exploring switching over too it.

Reply:
> nit: use round brackets instead of curly brackets.

Oops, that was left-over while I was still figuring out auto*.  Turns out I didn't need it.